vortex-array/src/expr/analysis/labeling.rs

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+use vortex_error::VortexExpect;
+use vortex_error::VortexResult;
+use vortex_utils::aliases::hash_map::HashMap;
+
+use crate::expr::Expression;
+use crate::expr::traversal::NodeExt;
+use crate::expr::traversal::NodeVisitor;
+use crate::expr::traversal::TraversalOrder;
+
+/// Label each node in an expression tree using a bottom-up traversal.
+///
+/// This function separates tree labeling into two distinct steps:
+/// 1. **Label Edge**: Compute a label for each node based only on the node itself
```

Comment:
Why is it called an edge if it is a label on the node, and not label for the connection between nodes?

Reply:
fixed this